timeline-chart/src/time-graph-unit-controller.ts

```diff
@@ -91,7 +95,15 @@ export class TimeGraphUnitController {
         if (this._viewRange.end > this.absoluteRange) {
             this._viewRange.end = this.absoluteRange;
         }
+        const newViewRangeLength = this.viewRangeLength;
+        this._scaleFactor = Number(prevViewRangeLength) / Number(newViewRangeLength);
```

Comment:
There is a property in the state controller called `zoomFactor` which has change handlers:
$$ZoomFactor = ZF = CanvasDisplayWidth/ViewRangeLength$$
The zoomFactor is updated every change to unitController.viewRange.  Could we move the logic for _scaleFactor into the stateController and move the handlers to onZoomChanged?  It would make more sense to only update the scale when the zoom changes.

Assuming canvasDisplayWidth is interchangeable with absoluteRange, the math works out very similarly:
In unit controller: $$scaleFactor = SF_{uc} = VR_{prev} / VR_{new}$$
In state controller: $$SF_{sc} = ZF_{prev} / ZF_{new} = (CDW_{prev}/VR_{prev}) /(CDW_{new}/VR_{new})$$
Since canvas display width won't change at the same time as view range:
$$CDW_{prev} = CDW_{new} \therefore SF_{sc} = SF_{uc}^{-1} = VR_{new} / VR_{prev}$$